includes/classes/observers/class.paypalwpp_incontext.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * Description of class.paypalwpp_incontext: This class is used to support order information related to PayPal Express In Context.  This way reduces the modifications of the various header files to nearly nothing in support of troubleshooting.
+ *
+ * @package paypalwpp
+ * @copyright Copyright 2003-2016 Zen Cart Development Team
+ * @copyright Portions Copyright 2003 osCommerce
+ * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ * @version $Id: mc12345678 Sat Aug 21 12:16:37 2016 -0500 Added in v1.6.0 $
+ *
+ * paypalwpp In Context  16-08-20 mc12345678
+ */
+class paypalwpp_incontext extends base {
+
+  
+  /*
+   * This is the observer for the PayPal return URIs when the PayPal payment is being processed at the end of the purchase.
+   */
+  function __construct() {
+    
+    $attachNotifier = array();
+
+    $attachNotifier[] = 'NOTIFY_HEADER_START_CHECKOUT_SHIPPING';
+    $attachNotifier[] = 'NOTIFY_HEADER_START_CHECKOUT_PAYMENT';
+
+    $this->attach($this, $attachNotifier);
+  }
+
+
+  // Clear the paypal_ec_token at the PayPal return URL (RETURNURL) when the
+  //  payment is cancelled at PayPal.  Without other process modifications, 
+  //  this is seen as necessary so that at the revisit to the payment 
+  //  confirmation page that PayPal Express In Context can again operate.
+  //    $attachNotifier[] = 'NOTIFY_HEADER_START_CHECKOUT_SHIPPING';
+  function updateNotifyHeaderStartCheckoutShipping(&$callingClass, $notifier) {
+    if (isset($_GET['ec_cancel']) && $_GET['ec_cancel'] == '1' 
+     && isset($_SESSION['paypal_ec_token']) && !empty($_SESSION['paypal_ec_token'])
+     && defined('MODULE_PAYMENT_PAYPALWPP_CHECKOUTSTYLE') && MODULE_PAYMENT_PAYPALWPP_CHECKOUTSTYLE == 'InContext' 
+     && defined('MODULE_PAYMENT_PAYPALWPP_MERCHANTID')    && MODULE_PAYMENT_PAYPALWPP_MERCHANTID    != '' 
+     && defined('MODULE_PAYMENT_PAYPALWPP_STATUS')        && MODULE_PAYMENT_PAYPALWPP_STATUS        == 'True') {
+      unset($GLOBALS[_SESSION]['paypal_ec_token']);
```

Comment:
I don't understand why you and others lately are referencing `$GLOBALS[_SESSION]` when $_SESSION is a superglobal and therefore `$_SESSION` is all that's needed.